src/components/formElements/ModalConfirmation.vue

```diff
@@ -0,0 +1,54 @@
+<template>
+  <ModalBasic
+    name="confirm"
+    :headerText="headerText"
+  >
+    <template v-slot:mait>
+      some text
+    </template>
+    <template v-slot:buttons>
+      <ButtonText
+        style-type="bordered"
+        text="confirm"
+        @click="submit"
+      />
+      <ButtonText
+        style-type="bordered"
+        text="cancel"
+        @click="cancel"
+      />
+    </template>
+  </ModalBasic>
+</template>
+
+<script>
+import ModalBasic from '@/components/formElements/ModalBasic.vue';
+import ButtonText from '@/components/formElements/ButtonText.vue';
+import { mapGetters } from 'vuex';
+
+export default {
+  components: {
+    ModalBasic,
+    ButtonText,
+  },
+  props: {
+    headerText: String,
+  },
+  computed: {
+    ...mapGetters(['isConfirmationModalShown']),
+  },
+  watch: {
+    isConfirmationModalShown() {
+      this.isConfirmationModalShown ? this.$modal.show('confirm') : this.$modal.hide('confirm');
+    },
```

Comment:
it can be a little bit strange to show or hide the component inside the component, don't you think?
And also, if we control it through the store, why do we send headerText through the prop?

Let's discuss how to make it more proper in the morning :)